### ISSUE TITLE
Show "loading" until JS ready

### ISSUE THREAD

**Issue by mitchhentges** [james-proxy/james#60]
It takes ~10 seconds to launch on Windows, and this is _with_ and SSD (though I doubt that disk-reading is the bottleneck).
On Linux, it takes around ~1 second to launch (which is still a little silly). Windows launch times should be comparable to Linux

**Comment by mitchhentges** [comment]
It appears to be first-launches that are ~10s. After the "allow network communications" firewall dialog has been dealt with, the startup time is similar to the other platforms.

**Comment by mitchhentges** [comment]
There isn't really much that can be done: if anything, it's probably on Electron's side.
As for the slow startup time on all platforms: the `requires` and `imports` run before the first render. It would be nice if the module loading could be deferred, so that initial renders can be done.

TL;DR: JS is slow, and requires all dependencies to be loaded from disk before the application will launch. Initial HTML page should show "loading", which will be replaced once JS is ready

**Comment by nerdbeere** [comment]
+1 for the "loading…" idea

**Comment by mitchhentges** [comment]
I made this:
![screenshot from 2016-01-28 15-37-29](https://cloud.githubusercontent.com/assets/7784737/12647482/b0212f5e-c5d5-11e5-90fd-082bdeb154bf.png)

But Electron doesn't even show the HTML + CSS while the JS is loading. I tested the Slack app, and it works by not showing the Slack window until it's completely ready. Maybe James will do the same thing, once installed as a regular app?

At any rate, I'm closing this issue, and it's a little out of our control
